File: lingting-core/src/main/java/live/lingting/framework/function/InterruptedRunnable.java
package live.lingting.framework.function;

import java.time.Duration;

/**
 * @author lingting 2024-01-26 15:34
 */
public interface InterruptedRunnable extends ThrowingRunnable {

	InterruptedRunnable THREAD_SLEEP = () -> Thread.sleep(500);

	static InterruptedRunnable threadSleep(Duration duration) {
		return () -> Thread.sleep(duration.toMillis());
	}

	@Override
	void run() throws InterruptedException;

}

File: lingting-core/src/main/kotlin/live/lingting/framework/function/InterruptedRunnable.kt
package live.lingting.framework.function

import java.time.Duration

/**
 * @author lingting 2024-01-26 15:34
 */
fun interface InterruptedRunnable : ThrowingRunnable {

    companion object {

        @JvmStatic
        fun threadSleep(duration: Duration): InterruptedRunnable {
            return InterruptedRunnable { Thread.sleep(duration.toMillis()) }
        }

        @JvmField
        val THREAD_SLEEP: InterruptedRunnable = InterruptedRunnable { Thread.sleep(50) }

    }

    @Throws(InterruptedException::class)
    override fun run()
}
